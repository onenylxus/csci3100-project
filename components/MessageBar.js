/**
 * CU There Team
 * @Component MessageBar -
 * @author Tse Shun Chi (1155125630)
 * @author Tsang Man Yi (1155125860)
 * @author Ng Wing Yin (1155127101)
 * @author Liu Hoi Pan (1155127464)
 * @author Lam Wai To Keith (1155133260)
 * Will be used by the AddInfoForm and EditProfileForm
 * @version 0.3.0
 */
/*
// Import
import React from 'react';
import { TextInput, TouchableOpacity, View, Text } from 'react-native';
import { FontAwesomeIcon } from '@fortawesome/react-native-fontawesome';
import { faPen } from '@fortawesome/free-solid-svg-icons';
import Style from '../assets/style';

// Export Message Bar
export default function MessageBar() {
  const [message, setMessage] = React.useState('');

  return (
    <View style={Style.messageBar}>
      <FontAwesomeIcon
        icon={faPen}
        size={15}
        style={{ marginHorizontal: 6, marginVertical: 12 }}
      />
      <TextInput
        placeholder="Write Message"
        style={{ width: '78%', alignSelf: 'center' }}
        value={message}
        onChangeText={(value) => setMessage(value)}
        enablesReturnKeyAutomatically
        multiline
      />
      <TouchableOpacity style={Style.sendMessageButtom}>
        <Text style={{ color: 'blue' }}>Send</Text>
      </TouchableOpacity>
    </View>
  );
}
*/
