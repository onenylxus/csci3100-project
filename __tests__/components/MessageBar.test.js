// Import
import React from 'react';
import { render } from '@testing-library/react-native';

// Import target component
import MessageBar from '../../components/MessageBar';

// Mock FontAwesome icons
jest.mock('@fortawesome/react-native-fontawesome', () => ({
  FontAwesomeIcon: '',
}));

// Run jest tests
describe('MessageBar', () => {
  let element;

  beforeEach(() => {
    // Render
    element = render(<MessageBar />);
  });

  it('matches snapshot', () => {
    expect(element.toJSON()).toMatchSnapshot();
  });
});
