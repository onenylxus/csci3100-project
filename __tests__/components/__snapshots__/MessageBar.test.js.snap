// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MessageBar matches snapshot 1`] = `
<View
  style={
    Object {
      "alignSelf": "center",
      "backgroundColor": "#fff",
      "borderColor": "#000",
      "borderRadius": 20,
      "borderWidth": 0.5,
      "flexDirection": "row",
      "margin": 10,
      "width": "95%",
    }
  }
>
  <
    icon={
      Object {
        "icon": Array [
          512,
          512,
          Array [],
          "f304",
          "M290.74 93.24l128.02 128.02-277.99 277.99-114.14 12.6C11.35 513.54-1.56 500.62.14 485.34l12.7-114.22 277.9-277.88zm207.2-19.06l-60.11-60.11c-18.75-18.75-49.16-18.75-67.91 0l-56.55 56.55 128.02 128.02 56.55-56.55c18.75-18.76 18.75-49.16 0-67.91z",
        ],
        "iconName": "pen",
        "prefix": "fas",
      }
    }
    size={15}
    style={
      Object {
        "marginHorizontal": 6,
        "marginVertical": 12,
      }
    }
  />
  <TextInput
    allowFontScaling={true}
    enablesReturnKeyAutomatically={true}
    multiline={true}
    onChangeText={[Function]}
    placeholder="Write Message"
    rejectResponderTermination={true}
    style={
      Object {
        "alignSelf": "center",
        "width": "78%",
      }
    }
    underlineColorAndroid="transparent"
    value=""
  />
  <View
    accessible={true}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignContent": "center",
        "justifyContent": "center",
        "marginHorizontal": 5,
        "opacity": 1,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "blue",
        }
      }
    >
      Send
    </Text>
  </View>
</View>
`;
